compiler/packages/babel-plugin-react-compiler/src/ReactiveScopes/PropagateScopeDependencies.ts

```diff
@@ -599,14 +607,22 @@ class Context {
      *  (all other decls e.g. `let x;` should be initialized in BuildHIR)
      */
     const originalDeclaration = this.#declarations.get(
-      maybeDependency.identifier.id,
+      maybeDependency.identifier.declarationId,
     );
     if (
       originalDeclaration !== undefined &&
       originalDeclaration.scope.value !== null
     ) {
       originalDeclaration.scope.each(scope => {
-        if (!this.#isScopeActive(scope.value)) {
+        if (
+          !this.#isScopeActive(scope.value) &&
+          // TODO LeaveSSA: key scope.declarations by DeclarationId
+          !Array.from(scope.value.declarations.values()).some(
+            decl =>
+              decl.identifier.declarationId ===
+              maybeDependency.identifier.declarationId,
+          )
```

Comment:
we likely need to key scope declarations by DeclarationId instead of IdentifierId but i need to think about this more

Reply:
I poked around a bit here and this makes sense to me.

This change means that `declarations`, `reassignments`, and `declarations` can record SSA-ified identifier instances for things like type inference (see `CollectReactiveIdentifiers` logic for pruned scope decls), or move `Place.reactive` -> `Identifier.reactive`. 

Is keying by `DeclarationId` primarily a constraint for codegen (i.e. the logic in RenameVariables)?